src/test_l1_dsFPD.c

```diff
@@ -609,39 +649,51 @@ void test_l1_dsFPD_negative_dsSetFPBlink (void)
  * |:--:|---------|----------|--------------|-----|
  * |01|Initialize with dsFPInit()| |dsERR_NONE|Ensure the system is initialized|
  * |02|Set all valid indicators to dsFPD_STATE_ON using dsSetFPState()|eIndicator: [Valid Indicator], state: dsFPD_STATE_ON |dsERR_NONE|Ensure the system is initialized|
- * |03|Call dsSetFPBrightness() and loop through all valid indicators from kIndicators|eIndicator: [Valid Indicator], eBrightness: 20+5x|dsERR_NONE|Brightness is set without any issues|
- * |04|Terminate with dsFPTerm()| |dsERR_NONE|Ensure the system is terminated|
+ * |03|Call dsSetFPBrightness() and loop through all valid indicators from the profile file |eIndicator: [Valid Indicator], eBrightness: 20+5x|dsERR_NONE|Brightness is set without any issues|
+ * |04|Set all valid indicators to dsFPD_STATE_OFF using dsSetFPState()|eIndicator: [Valid Indicator], state: dsFPD_STATE_OFF |dsERR_OPERATION_NOT_SUPPORTED|Validate invalid parameter handling for eIndicator|
+ * |05|Terminate with dsFPTerm()| |dsERR_NONE|Ensure the system is terminated|
  * 
- * @note Valid indicators can retrieved from id element in kIndicators in the dsFPDSettings.h file
+ * @note Valid indicators can retrieved from yaml file
  * 
  */
 void test_l1_dsFPD_positive_dsSetFPBrightness (void)
 {
     gTestID = 9;
     UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
     dsError_t result;
+    dsFPDIndicator_t eIndicator;
+    int count = 0;
+    char buffer[DS_FPD_KEY_SIZE];
 
     // Step 01: Initialize with dsFPInit()
     result = dsFPInit();
     UT_ASSERT_EQUAL(result, dsERR_NONE);
 
     // Step 02: Set all valid indicators to dsFPD_STATE_ON using dsSetFPState()
-    // Step 03: Call dsSetFPBrightness() and loop through all valid indicators from kIndicators
-    for (int i = 0; i < sizeof(kIndicators) / sizeof(kIndicators[0]); ++i)
+    // Step 03: Call dsSetFPBrightness() and loop through all valid indicators from yaml file
+    // Step 04: Set all valid indicators to dsFPD_STATE_OFF using dsSetFPState()
+    // Retrieve the number of valid indicators using the macro
+    count = UT_KVP_PROFILE_GET_UINT32("dsFPD/Number_of_Indicators");
+    for (int i = 0; i < count; i++)
     {
-        UT_LOG("\n In %s , Indicator: [%d]\n", __FUNCTION__, kIndicators[i].id);
-        result = dsSetFPState(kIndicators[i].id, dsFPD_STATE_ON);
+        snprintf(buffer, DS_FPD_KEY_SIZE, "dsFPD/SupportedFPDIndicators/%d/Indicator_Type", i);
+        eIndicator = UT_KVP_PROFILE_GET_UINT32(buffer);
+        UT_LOG("\n In %s , Indicator: [%d]\n", __FUNCTION__, eIndicator);
+        result = dsSetFPState(eIndicator, dsFPD_STATE_ON);
         UT_ASSERT_EQUAL(result, dsERR_NONE);
         
-        result = dsSetFPBrightness(kIndicators[i].id, 0);
+        result = dsSetFPBrightness(eIndicator, 0);
```

Comment:
now brightness is part of profile file, so we can read the Max, Min values from the profile.. and add a average value (max+min)/2

MAX_BRIGHTNESS : 100            # Maximum brightness value of FPD LEDs
        MIN_BRIGHTNESS : 0              # Minimum brightness value of FPD LEDs

Reply:
Done